### PR TITLE
fix(shwap/bitswap): Blockstore.GetSize: getting size with no compute

### PR DESCRIPTION
**Pull request by Wondertan**
Bitswap prioritizes peers based on their active/pending work and the priority that peers set for requests(work) themselves. The prioritization happens on the `Get` operation of `Blockstore` not `GetSize`. However, we currently do the most expensive work in `GetSize` followed by `Get`. It is still part of the same WANT_BLOCK request(since #3813), and `GetSize` work is usually cached for `Get` to catch it. Still, the prioritization makes it so that the time between `GetSize` and `Get` can be pretty long, inducing cache misses and redoing the same work again, which is confirmed by profiles. Also, doing the expensive part in `GetSize` avoids Bitswap server's rate limiting. 

All this brings the need to make `GetSize` as lightweight as possible, leaving actual proof computed to `Get`. This PR achieves this by defining a constant max block size for every type block.

--- 

🎱 mb blonks certified

### CODE REVIEW

**Comment by cristaloleg** [comment]
Lint fail https://github.com/celestiaorg/celestia-node/actions/runs/11556547784/job/32164506582?pr=3894#step:4:31

**Comment by walldiss** [comment]
If GetSize is used only for prioritisation of bitswap server hanling of requests, perhaps don't really need to calculate sizes. Alternative could be to return const values depending on type of the cid and our estimate of proper priority / load of handling. For examples:
- samples: 100
- rows: 10000 (much heavier and less important than namespace data)
- namespace data: 5000 

This would allow us to remove the need for heavy reads to get the priority for serving of requests and will easy the pressure on cache.

**Comment by Wondertan** [comment]
In sync, we agreed to keep the Size computation based on real blocksize to be safe. Making all sizes fixed to max block size may limit bitswap's throughput for small block sizes.

File: share/shwap/p2p/bitswap/sample_block.go [hunk 1]
```diff
@@ -71,6 +74,12 @@ func (sb *SampleBlock) Height() uint64 {
 	return sb.ID.Height
 }
 
+func (sb *SampleBlock) Size(ctx context.Context, acc eds.Accessor) (int, error) {
+	squareSize := acc.Size(ctx)
+	sampleSize := libshare.ShareSize + share.AxisRootSize*int(math.Log2(float64(squareSize)))
```
Comment (renaynay) [inline share/shwap/p2p/bitswap/sample_block.go:79]: is there no helper util we can use to determine proof size for a share based on dah?

**Review by renaynay — COMMENTED** [review]
Do you need additional tests for this? Are the changes already covered by some test suite?

File: share/shwap/p2p/bitswap/sample_block.go (hunk 1 above)
Comment (Wondertan) [reply share/shwap/p2p/bitswap/sample_block.go:79]: I am not aware of such

**Comment by Wondertan** [comment]
It is covered

**Review by walldiss — DISMISSED** [review]

**Review by vgonkivs — DISMISSED** [review]

**Comment by Wondertan** [comment]
Disabling the automerge for now. I want to do another change here

**Review by renaynay — APPROVED** [review]
utack

**Review by walldiss — APPROVED** [review]

**Review by renaynay — APPROVED** [review]

**Review by walldiss — APPROVED** [review]

**Pull request merged**